scripts/deploy.ts

```diff
@@ -1,122 +1,194 @@
-      break;
     // opeth
     case 10:
     // topeth
     case 11155420:
-      walletImplementationContractName = 'WalletSimple';
-      forwarderContractName = 'Forwarder';
-      forwarderFactoryContractName = 'ForwarderFactory';
-      contractPath = `contracts/${walletImplementationContractName}.sol:${walletImplementationContractName}`;
-      break;
+      return [
+        'WalletSimple',
+        'WalletFactory',
+        'Forwarder',
+        'ForwarderFactory',
+        'contracts/WalletSimple.sol:WalletSimple'
+      ];
     // zketh
     case 324:
     // tzketh
     case 300:
-      walletImplementationContractName = 'ZkethWalletSimple';
-      contractPath = `contracts/coins/${walletImplementationContractName}.sol:${walletImplementationContractName}`;
-      break;
+      return [
+        'ZkethWalletSimple',
+        'WalletFactory',
+        'Forwarder',
+        'ForwarderFactory',
+        'contracts/ZkethWalletSimple.sol:ZkethWalletSimple'
+      ];
+    default:
+      return ['', '', '', '', ''];
   }
+}
 
-  console.log(
-    'Deployed wallet contract called: ' + walletImplementationContractName
-  );
+async function main() {
+  const output = {
+    walletImplementation: '',
+    walletFactory: '',
+    forwarderImplementation: '',
+    forwarderFactory: ''
+  };
+  const version = process.env.VERSION ? process.env.VERSION.split('.')[0] : '';
 
-  const WalletSimple = await ethers.getContractFactory(
-    walletImplementationContractName
-  );
-  const walletSimple = await WalletSimple.deploy(gasParams);
-  await walletSimple.deployed();
-  output.walletImplementation = walletSimple.address;
-  console.log('WalletSimple deployed at ' + walletSimple.address);
+  const feeData = await ethers.provider.getFeeData();
+  const gasParams = {
+    gasPrice: version === 'v1' ? feeData.gasPrice!.mul('2') : feeData.gasPrice
+  };
 
-  const WalletFactory = await ethers.getContractFactory(
-    walletFactoryContractName
+  let walletDeployer, forwarderDeployer;
+  if (version === 'v1') {
+    [walletDeployer, forwarderDeployer] = await ethers.getSigners();
+  } else {
+    [forwarderDeployer] = await ethers.getSigners();
+  }
+  const chainId = 1;
+
+  const [
+    walletImplementationContractName,
+    walletFactoryContractName,
+    forwarderImplementationContractName,
+    forwarderFactoryContractName,
+    contractPath
+  ] = getContractNames(version, chainId);
+
+  // Wallet Contract Deployment
+  if (version === 'v1' && walletDeployer) {
+    await selfTransferForV1(walletDeployer, gasParams.gasPrice, 2);
+  }
+  const WalletImplementation =
+    version === 'v1'
+      ? await ethers.getContractFactory(
+          walletImplementationContractName,
+          walletDeployer
+        )
+      : await ethers.getContractFactory(walletImplementationContractName);
+
+  console.log(`Deploying ${walletImplementationContractName}....`);
```

Comment:
skip the wallet and forwarder deployment if its already deployed

Reply:
Done based on the `Signer.getTransactionCount()`